### ISSUE TITLE
0.8.3/0.8.4 rpc out-of-service 

### ISSUE THREAD

**Issue by bCoder778** [Qitmeer/qitmeer#217]
[qitmeerd.log](https://github.com/Qitmeer/qitmeer/files/3949234/qitmeerd.log)

multiple nodes cannot be accessed at the same time

**Comment by lochjin** [comment]
> [qitmeerd.log](https://github.com/Qitmeer/qitmeer/files/3949234/qitmeerd.log)
> 
> multiple nodes cannot be accessed at the same time

Please use qitmeer0.8.4-dev

**Comment by dindinw** [comment]
reopen the issue, since `0.8.4` also share the same problem.

**Comment by blocklee** [comment]
[qitmeer.log](https://github.com/Qitmeer/qitmeer/files/3982126/qitmeer.log)

I have  the same problem, but I can not stop the qitmeer when using 'ctrl+c'. 

miner error:

`[ERROR] error http response                 module=miner status="503 Service Unavailable" body="503 Too busy.  Try again later.\n"`

**Comment by blocklee** [comment]
本地节点出现以下信息，同时节点暂停同步：

```
2019-12-19|20:12:53.735 [[33mWARN [0m] RPC authentication failure          [33mfrom[0m=127.0.0.1:60058
2019-12-19|20:17:39.130 [[33mWARN [0m] RPC authentication failure          [33mfrom[0m=127.0.0.1:60165
2019-12-19|20:19:48.858 [[33mWARN [0m] RPC authentication failure          [33mfrom[0m=127.0.0.1:60255
2019-12-19|20:20:54.732 [[33mWARN [0m] RPC authentication failure          [33mfrom[0m=127.0.0.1:60299
2019-12-20|00:49:23.958 [[32mINFO [0m] Your synchronization has been completed.  [32mmodule[0m=blkmanager
2019-12-20|00:49:23.982 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.15.132.180:18130 (outbound) 
2019-12-20|00:49:23.988 [[32mINFO [0m] Lost peer                           [32mmodule[0m=blkmanager [32mpeer[0m="103.29.70.78:18130 (outbound)"
2019-12-20|00:49:23.991 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 47.88.220.44:18130 (outbound) 
2019-12-20|00:49:23.994 [[32mINFO [0m] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.001 [[32mINFO [0m] Syncing to state (13502,10834,10834,13557,1) from peer 103.29.70.78:18130 cur graph state:(13211,10582,10582,13212,1) [32mmodule[0m=blkmanager
2019-12-20|00:49:23.998 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.15.132.180:18130 (outbound) 
2019-12-20|00:49:24.010 [[32mINFO [0m] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.016 [[32mINFO [0m] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.012 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 47.88.220.44:18130 (outbound) 
2019-12-20|00:49:24.021 [[32mINFO [0m] Lost peer                           [32mmodule[0m=blkmanager [32mpeer[0m="103.15.132.180:18130 (outbound)"
2019-12-20|00:49:24.025 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.15.132.180:18130 (outbound) 
2019-12-20|00:49:24.029 [[32mINFO [0m] Lost peer                           [32mmodule[0m=blkmanager [32mpeer[0m="47.88.220.44:18130 (outbound)"
2019-12-20|00:49:24.033 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.29.70.78:18130 (outbound) 
2019-12-20|00:49:24.038 [[32mINFO [0m] New valid peer: 47.88.220.44:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.041 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 47.88.220.44:18130 (outbound) 
2019-12-20|00:49:24.045 [[32mINFO [0m] New valid peer: 103.15.132.180:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.048 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.29.70.78:18130 (outbound) 
2019-12-20|00:49:24.054 [[32mINFO [0m] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.057 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.29.70.78:18130 (outbound) 
2019-12-20|00:49:24.062 [[32mINFO [0m] New valid peer: 103.15.132.180:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.066 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.29.70.78:18130 (outbound) 
2019-12-20|00:49:24.071 [[32mINFO [0m] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.073 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 47.88.220.44:18130 (outbound) 
2019-12-20|00:49:24.078 [[32mINFO [0m] New valid peer: 103.15.132.180:18130 (outbound),user-agent:qitmeer:0.8.4 [32mmodule[0m=blkmanager
2019-12-20|00:49:24.082 [[32mINFO [0m] Max peers reached [125] - disconnecting peer 103.15.132.180:18130 (outbound) 
2019-12-20|00:49:24.087 [[32mINFO [0m] New valid peer: 103.15.132.180:18130 (outbound),user-agent:qitmeer:0.8.4 

```

**Comment by blocklee** [comment]
节点报错 RPC clients exceeded ，并且重启节点时自动终止：
`[ERROR] Uable to start server               error="no block at order 20620 exists"`

```
2019-12-20|15:14:18.827 [DEBUG] Received pong from 103.15.132.180:18130 
2019-12-20|15:14:20.321 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45601
2019-12-20|15:14:25.489 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45608
2019-12-20|15:14:30.662 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45613
2019-12-20|15:14:35.832 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45618
2019-12-20|15:14:37.107 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45620
2019-12-20|15:14:41.001 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45624
2019-12-20|15:14:46.175 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45635
2019-12-20|15:14:51.344 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45641
2019-12-20|15:14:56.514 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45647
2019-12-20|15:14:56.825 [DEBUG] Received ping from 47.103.194.115:18130 
2019-12-20|15:14:56.825 [DEBUG] Adding deadline for command pong for peer 47.103.194.115:18130 
2019-12-20|15:14:56.825 [DEBUG] Sending pong to 47.103.194.115:18130 
2019-12-20|15:14:56.825 [DEBUG] Sending ping to 47.103.194.115:18130 
2019-12-20|15:14:56.825 [DEBUG] Adding deadline for command ping for peer 47.103.194.115:18130 
2019-12-20|15:14:56.992 [DEBUG] Received pong from 47.103.194.115:18130 
2019-12-20|15:15:01.688 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45658
2019-12-20|15:15:06.858 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45668
2019-12-20|15:15:12.028 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45673
2019-12-20|15:15:17.199 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45679
2019-12-20|15:15:22.368 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45690
2019-12-20|15:15:27.536 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45695
2019-12-20|15:15:32.706 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45704
2019-12-20|15:15:34.310 [DEBUG] Sending ping to 47.88.220.44:18130 
2019-12-20|15:15:34.310 [DEBUG] Adding deadline for command ping for peer 47.88.220.44:18130 
2019-12-20|15:15:34.311 [DEBUG] Received pong from 47.88.220.44:18130 
2019-12-20|15:15:34.313 [DEBUG] Received ping from 47.88.220.44:18130 
2019-12-20|15:15:34.313 [DEBUG] Adding deadline for command pong for peer 47.88.220.44:18130 
2019-12-20|15:15:34.313 [DEBUG] Sending pong to 47.88.220.44:18130 
2019-12-20|15:15:37.875 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45712
2019-12-20|15:15:43.044 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45716
2019-12-20|15:15:48.212 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45725
2019-12-20|15:15:53.382 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45731
2019-12-20|15:15:54.610 [DEBUG] Sending ping to 103.29.70.78:18130 
2019-12-20|15:15:54.610 [DEBUG] Adding deadline for command ping for peer 103.29.70.78:18130 
2019-12-20|15:15:58.555 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45735
2019-12-20|15:16:03.727 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45744
2019-12-20|15:16:08.861 [DEBUG] Sending ping to 47.93.20.102:18130 
2019-12-20|15:16:08.861 [DEBUG] Adding deadline for command ping for peer 47.93.20.102:18130 
2019-12-20|15:16:08.897 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45751
2019-12-20|15:16:14.067 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45754
2019-12-20|15:16:16.439 [INFO ] Shutting down...                    Received signal=interrupt
2019-12-20|15:16:16.439 [INFO ] Gracefully shutting down the server... 
2019-12-20|15:16:16.439 [INFO ] Stopping Server                     module=node
2019-12-20|15:16:16.439 [DEBUG] RPC Server is stopping 
2019-12-20|15:16:16.439 [WARN ] Stopping P2P Server 
2019-12-20|15:16:16.439 [WARN ] wait for P2P stop ... 
2019-12-20|15:16:16.439 [INFO ] Connection manager stopped 
2019-12-20|15:16:16.439 [INFO ] Address manager shutting down 
2019-12-20|15:16:16.439 [WARN ] P2P Server stopped 
2019-12-20|15:16:16.439 [DEBUG] Stopping Qitmeer full node service  module=node
2019-12-20|15:16:16.439 [INFO ] try stop bm                         module=node
2019-12-20|15:16:16.439 [INFO ] Block manager shutting down         module=blkmanager
2019-12-20|15:16:16.439 [INFO ] Wait For Block manager stop ...     module=blkmanager
2019-12-20|15:16:19.235 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45770
2019-12-20|15:16:24.405 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45777
2019-12-20|15:16:28.366 [INFO ] Already shutting down...            Received signal=interrupt
2019-12-20|15:16:29.572 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45782
2019-12-20|15:16:34.740 [INFO ] RPC clients exceeded                max=10 client=1.203.65.55:45786
2019-12-20|15:17:08.369 [INFO ] System info                         Qitmeer Version=0.8.4+dev-8b58578 Go version=go1.12.7
2019-12-20|15:17:08.369 [INFO ] System info                         UUID=d8c0a80e-9313-4335-8e8a-0d4c47a081d0
2019-12-20|15:17:08.369 [INFO ] System info                         Home dir=/root/.qitmeerd
2019-12-20|15:17:08.369 [INFO ] Loading block database              dbPath=/root/.qitmeerd/data/testnet/blocks_ffldb
2019-12-20|15:17:08.397 [INFO ] Detected unclean shutdown - Repairing... module=database
2019-12-20|15:17:08.397 [DEBUG] Metadata claims file 0, offset 10461584. Block data is at file 0, offset 10553032 module=database
2019-12-20|15:17:08.397 [DEBUG] ROLLBACK: Rolling back to           module=database file num=0 offset=10461584
2019-12-20|15:17:08.401 [INFO ] Database sync complete              module=database
2019-12-20|15:17:08.401 [INFO ] Block database loaded 
2019-12-20|15:17:08.401 [DEBUG] Skipping local address              error="address 127.0.0.1 is not routable"
2019-12-20|15:17:08.401 [DEBUG] Skipping local address              error="address 172.21.210.141 is not routable"
2019-12-20|15:17:08.401 [DEBUG] Skipping local address              error="address 172.17.0.1 is not routable"
2019-12-20|15:17:08.402 [DEBUG] Register service to node            module=node     service="{service:qitmeer initFunc:0xaa7790}"
2019-12-20|15:17:08.402 [INFO ] Starting Server                     module=node
2019-12-20|15:17:08.412 [INFO ] Transaction index is enabled        module=node
2019-12-20|15:17:08.413 [INFO ] anticone size:6                     module=blockdag
2019-12-20|15:17:08.413 [INFO ] Loading dag ...                     module=blockchain
2019-12-20|15:17:09.171 [INFO ] Dag loaded:loadTime=757.299184ms    module=blockchain
2019-12-20|15:17:10.082 [DEBUG] Current internal block              block id=20620
2019-12-20|15:17:10.082 [DEBUG] Current transaction index tip       order=20619 hash=e6d2db5335ea4af5b555d84140390638af0985321d7942d98f6a6af35688e2e6
2019-12-20|15:17:10.082 [INFO ] Catching up indexes from order 20619 to 4294967295 
2019-12-20|15:17:10.082 [ERROR] Uable to start server               error="no block at order 20620 exists"
2019-12-20|15:17:10.082 [INFO ] Gracefully shutting down the server... 
2019-12-20|15:17:10.083 [INFO ] Stopping Server                     module=node
2019-12-20|15:17:10.083 [WARN ] Stopping P2P Server 
2019-12-20|15:17:10.083 [WARN ] wait for P2P stop ... 
2019-12-20|15:17:10.083 [WARN ] P2P Server stopped 
2019-12-20|15:17:10.083 [INFO ] Waiting for server shutdown         module=node
2019-12-20|15:17:10.083 [INFO ] Gracefully shutting down the database... 
2019-12-20|15:17:10.097 [INFO ] Shutdown complete 
2019-12-20|15:17:56.361 [INFO ] System info                         Qitmeer Version=0.8.4+dev-8b58578 Go version=go1.12.7
2019-12-20|15:17:56.361 [INFO ] System info                         UUID=506192ea-2a45-4840-ae55-2375db9fc813
2019-12-20|15:17:56.361 [INFO ] System info                         Home dir=/root/.qitmeerd
2019-12-20|15:17:56.361 [INFO ] Loading block database              dbPath=/root/.qitmeerd/data/testnet/blocks_ffldb
2019-12-20|15:17:56.371 [INFO ] Block database loaded 
2019-12-20|15:17:56.372 [DEBUG] Skipping local address              error="address 127.0.0.1 is not routable"
2019-12-20|15:17:56.372 [DEBUG] Skipping local address              error="address 172.21.210.141 is not routable"
2019-12-20|15:17:56.372 [DEBUG] Skipping local address              error="address 172.17.0.1 is not routable"
2019-12-20|15:17:56.373 [DEBUG] Register service to node            module=node service="{service:qitmeer initFunc:0xaa7790}"
2019-12-20|15:17:56.373 [INFO ] Starting Server                     module=node
2019-12-20|15:17:56.374 [INFO ] Transaction index is enabled        module=node
2019-12-20|15:17:56.374 [INFO ] anticone size:6                     module=blockdag
2019-12-20|15:17:56.375 [INFO ] Loading dag ...                     module=blockchain
2019-12-20|15:17:57.210 [INFO ] Dag loaded:loadTime=835.387474ms    module=blockchain
2019-12-20|15:17:58.210 [DEBUG] Current internal block              block id=20620
2019-12-20|15:17:58.210 [DEBUG] Current transaction index tip       order=20619 hash=e6d2db5335ea4af5b555d84140390638af0985321d7942d98f6a6af35688e2e6
2019-12-20|15:17:58.210 [INFO ] Catching up indexes from order 20619 to 4294967295 
2019-12-20|15:17:58.210 [ERROR] Uable to start server               error="no block at order 20620 exists"
2019-12-20|15:17:58.210 [INFO ] Gracefully shutting down the server... 
2019-12-20|15:17:58.210 [INFO ] Stopping Server                     module=node
2019-12-20|15:17:58.210 [WARN ] Stopping P2P Server 
2019-12-20|15:17:58.210 [WARN ] wait for P2P stop ... 
2019-12-20|15:17:58.210 [WARN ] P2P Server stopped 
2019-12-20|15:17:58.210 [INFO ] Waiting for server shutdown         module=node
2019-12-20|15:17:58.210 [INFO ] Gracefully shutting down the database... 
2019-12-20|15:17:58.246 [INFO ] Shutdown complete 
```

**Comment by forchain** [comment]
[data.zip](https://github.com/Qitmeer/qitmeer/files/3987411/data.zip)
@lochjin please use this data to reproduce

**Comment by dindinw** [comment]
@bCoder778 @blocklee @forchain issue handled by #241, Please verify it has been fixed.

**Comment by forchain** [comment]
> @bCoder778 @blocklee @forchain issue handled by #241, Please verify it has been fixed.

verified

**Comment by dindinw** [comment]
fixed by #241
